.github/workflows/precommit.yml

```diff
@@ -5,6 +5,7 @@ on:
   push:
     branches:
       - master
+  workflow_dispatch:
```

Comment:
Do we really need this? What are the use cases?

Reply:
This is for when the last commit has a `[skip ci]` flag set so the workflow can be manually triggered (as it is required for PR merge)